### PR TITLE
Release 3.0.6 release notes.

### PR DESCRIPTION
**Pull request by awholegunch**
Passes [Quick Build](http://builds.cask.co/browse/CDAP-DQBP32-2).

[Release Notes](http://builds.cask.co/artifact/CDAP-DQBP32/shared/build-2/Docs-HTML/3.0.5/en/reference-manual/release-notes.html#release-3-0-6)

### CODE REVIEW

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 1]
```diff
@@ -24,6 +24,45 @@ Cask Data Application Platform Release Notes
...
+
+- Fixed a problem where a Hadoop filesystem object was being instantiated before the
+  Kerberos keytab login was completed, leading to CDAP processes failing after the initial
+  ticket expired (`CDAP-3660 <https://issues.cask.co/browse/CDAP-3660>`__).
+
+- Fixed a problem where not all dependent services were being stopped when a system service was stopped 
+  (`CDAP-3861 <https://issues.cask.co/browse/CDAP-3861>`__).
+
+- Fixed a problem with the CDAP Router not have a configurable timeout for connections 
+  (`CDAP-3887 <https://issues.cask.co/browse/CDAP-3887>`__).
```
Comment (sagarkapare) [inline cdap-docs/reference-manual/source/release-notes.rst]: This needs to be removed from the list. We are not shipping it with 3.0.6

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 2]
```diff
@@ -24,6 +24,45 @@ Cask Data Application Platform Release Notes
...
 
 
+`Release 3.0.6 <http://docs.cask.co/cdap/3.0.6/index.html>`__
+=============================================================
+
+Bug Fixes
+---------
+
+- Fixed a problem with the CDAP Master restarting the Twill app indefinitely in certain circumstances
+  (`CDAP-1543 <https://issues.cask.co/browse/CDAP-1543>`__).
```
Comment (sagarkapare) [inline cdap-docs/reference-manual/source/release-notes.rst]: This bug is marked as won't fix. should be removed from the list.

**Comment by jwang47** [comment]
Please also add the release note from https://issues.cask.co/browse/CDAP-3890 -- I just added it now.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 1 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Fixed.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 2 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Fixed.

**Comment by poornachandra** [comment]
Since we are upgrading Tephra from 0.4.2 to 0.6.3, should we mention it in the release notes?

**Comment by awholegunch** [comment]
@jwang47 : Added.
@poornachandra : We should, and I will.

**Comment by awholegunch** [comment]
[Quick Build](http://builds.cask.co/browse/CDAP-DQBP32-3)

[Release Notes](http://builds.cask.co/artifact/CDAP-DQBP32/shared/build-3/Docs-HTML/3.0.5/en/reference-manual/release-notes.html#release-3-0-6)

**Comment by awholegunch** [comment]
@poornachandra @jwang47 @sagarkapare I've updated this; please take another look.

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 3]
```diff
@@ -24,6 +24,46 @@ Cask Data Application Platform Release Notes
...
+- Fixed a problem where a Hadoop filesystem object was being instantiated before the
+  Kerberos keytab login was completed, leading to CDAP processes failing after the initial
+  ticket expired (`CDAP-3660 <https://issues.cask.co/browse/CDAP-3660>`__).
+
+- Fixed a problem where not all dependent services were being stopped when a system service was stopped 
+  (`CDAP-3861 <https://issues.cask.co/browse/CDAP-3861>`__).
+
+- Fixed a problem with dataset performance degradation due to the making of 
+  multiple remote calls for each "get dataset" request 
+  (`CDAP-3890 <https://issues.cask.co/browse/CDAP-3890>`__).
```
Comment (jwang47) [inline cdap-docs/reference-manual/source/release-notes.rst]: Just moved release note to https://issues.cask.co/browse/CDAP-3902 where it belongs.

**Comment by sagarkapare** [comment]
@awholegunch can you please add https://issues.cask.co/browse/CDAP-4133 to the release notes?

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 3 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: What is https://issues.cask.co/browse/CDAP-3890 then? It looks identical to me, but maybe I'm missing something?

Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Which of the two (or should I reference both?) should be in the release notes, is what I'm asking.

Comment (jwang47) [inline cdap-docs/reference-manual/source/release-notes.rst]: 3890 is the parent JIRA, which is for the performance degradation issue, whereas 3902 is the fix we have for 3.0-3.3. The current fix, however, removes the namespace check when getting a dataset, so we'd like to add that back in later. Once we add the namespace check back in, we can resolve 3890.

Comment (jwang47) [inline cdap-docs/reference-manual/source/release-notes.rst]: So, we should reference 3902 only.

Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Thanks. Fixed.

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 4]
```diff
@@ -24,6 +24,49 @@ Cask Data Application Platform Release Notes
...
+- Fixed problems with the installation documentation 
+  (`CDAP-4093 <https://issues.cask.co/browse/CDAP-4093>`__).
+
+- Fixed an issue with the incorrect labelling, after failover, of programs that had been running prior to failover 
+  (`CDAP-4119 <https://issues.cask.co/browse/CDAP-4119>`__).
+
+New Features
+------------
+
+- Updated the version of `Tephra <http://tephra.io>`__ included in CDAP to v0.6.3
```
Comment (poornachandra) [inline cdap-docs/reference-manual/source/release-notes.rst]: used in CDAP, right?

Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Sure.

**Comment by awholegunch** [comment]
@sagar : Can you take another look and let me know if this is finished?

**Comment by sagar** [comment]
@awholegunch you got the wrong Sagar, bud :)

**Comment by sagarkapare** [comment]
@awholegunch Please put this PR on hold till we figure out what version of the Tephra we release the 3.0.6 with.

**Comment by awholegunch** [comment]
@sagarkapare I've updated the release notes with respect to the Tephra version. Please take a look.

**Comment by sagarkapare** [comment]
@awholegunch Please add https://issues.cask.co/browse/CDAP-3563 to the list. Rest looks good!

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 5]
```diff
@@ -24,6 +24,48 @@ Cask Data Application Platform Release Notes
...
+- Fixed problems with the installation documentation 
+  (`CDAP-4093 <https://issues.cask.co/browse/CDAP-4093>`__).
+
+- Fixed an issue with the incorrect labelling, after failover, of programs that had been running prior to failover 
+  (`CDAP-4119 <https://issues.cask.co/browse/CDAP-4119>`__).
+
+New Features
+------------
+
+- Updated the version of `Tephra <http://tephra.io>`__ used in CDAP to v0.4.4.
```
Comment (sagarkapare) [inline cdap-docs/reference-manual/source/release-notes.rst]: should we add the bug number here - https://issues.cask.co/browse/CDAP-4181?

**Comment by awholegunch** [comment]
Passes [Quick Build](http://builds.cask.co/browse/CDAP-DQBP32-7).

[Release Notes](http://builds.cask.co/artifact/CDAP-DQBP32/shared/build-7/Docs-HTML/3.0.5/en/reference-manual/release-notes.html#release-3-0-6)

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 5 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Sure. I had the other one off because it was confusing.

Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Done.

**Comment by sagarkapare** [comment]
LGTM :+1:

**Comment by awholegunch** [comment]
Passes [Quick Build](http://builds.cask.co/browse/CDAP-DQBP32-8).

[Release Notes](http://builds.cask.co/artifact/CDAP-DQBP32/shared/build-8/Docs-HTML/3.0.5/en/reference-manual/release-notes.html#release-3-0-6)

**Pull request merged**